lightning/src/ln/channelmanager.rs

```diff
@@ -926,13 +926,13 @@ impl ClaimablePayment {
 	}
 
 	/// Returns the inbound `channel_id`s for all HTLCs associated with the payment.
-	fn get_channel_ids(&self) -> Vec<Option<ChannelId>> {
-		self.htlcs.iter().map(|htlc| Some(htlc.prev_hop.channel_id)).collect()
+	fn get_channel_ids(&self) -> Vec<ChannelId> {
+		self.htlcs.iter().map(|htlc| htlc.prev_hop.channel_id).collect()
 	}
 
 	/// Returns the inbound `user_channel_id`s for all HTLCs associated with the payment.
-	fn get_user_channel_ids(&self) -> Vec<Option<u128>> {
-		self.htlcs.iter().map(|htlc| htlc.prev_hop.user_channel_id).collect()
+	fn get_user_channel_ids(&self) -> Vec<u128> {
+		self.htlcs.iter().map(|htlc| htlc.prev_hop.user_channel_id.unwrap_or(0)).collect()
```

Comment:
Would be nice to `filter_map` instead of falling back to inserting `0`, could also document on the method(s) for what versions the lists will be inaccurate, if any.

Reply:
That looks cleaner! Updated in [pr3655.03](https://github.com/shaavan/rust-lightning/commits/pr3655.03)
Also, I have added a small note that the list will be incomplete for HTLC created using LDK versions `<= 0.0.112`

Thanks a lot for the suggestion!